### PR TITLE
GUI-2854: Support bucket policy configuration

### PR DESCRIPTION
**Pull request by kamalgill**
https://eucalyptus.atlassian.net/browse/GUI-2854

### CODE REVIEW

File: eucaconsole/static/js/pages/bucket_details.js [hunk 1]
```diff
@@ -155,7 +220,10 @@ angular.module('BucketDetailsPage', ['S3SharingPanel', 'EucaConsoleUtils', 'Cors
                         $scope.savingCorsConfig = true;
                         $scope.corsError = '';
                         var csrfToken = angular.element('#csrf_token').val();
-                        CorsService.setCorsConfig($scope.bucketName, csrfToken, $scope.codeEditor.getValue())
+                        var corsTextarea = angular.element('#cors-textarea');
+                        // Fall back to standard textarea in cases where CodeMirror fails to initialize
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: Is this likely and how should we address this elsewhere?

File: eucaconsole/static/js/pages/bucket_details.js [hunk 2]
```diff
@@ -166,6 +234,7 @@ angular.module('BucketDetailsPage', ['S3SharingPanel', 'EucaConsoleUtils', 'Cors
                                 $scope.savingCorsConfig = false;
                             });
                     };
+                    // FIXME: modal:open doesn't fire in Angular 1.5
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: Is there a ticket for this?

File: eucaconsole/static/sass/pages/bucket_details.scss [hunk 3]
```diff
@@ -23,21 +23,25 @@
     }
 }
 
-// CORS Configuration modal dialog
-.cors-config-modal {
+// CORS Configuration and Bucket Policy modal dialogs
+.cors-config-modal, #bucket-policy-modal {
```
Comment (dkavanagh) [inline eucaconsole/static/sass/pages/bucket_details.scss]: Curious why use class on one and id on the other?

File: eucaconsole/static/js/pages/bucket_details.js (hunk 2 above)
Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: This is similar to the Angular 1.5 issue we've seen elsewhere (e.g. CloudWatch charts dialog on resource monitoring pages).  Do you want a distinct ticket for that, or is the CW charts one adequate?

File: eucaconsole/static/js/pages/bucket_details.js (hunk 1 above)
Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: CodeMirror isn't initialized due to the Angular 1.5 modal:open issue.

File: eucaconsole/static/js/pages/bucket_details.js (hunk 2 above)
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: What I saw in the debugger was that modal:open was being fired, but the listener wasn't set up in time. Have you confirmed that ordering issue?

File: eucaconsole/static/js/pages/bucket_details.js (hunk 1 above)
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: I hate to have a band-aid for another issue that needs to be solved, right?

File: eucaconsole/static/js/pages/bucket_details.js (hunk 2 above)
Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: Yes, and it seems to be an issue (as it is everywhere else) with our custom modal.  I haven't found a fix yet -- any luck with the CloudWatch modal issue?

File: eucaconsole/static/js/pages/bucket_details.js (hunk 1 above)
Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: This is a temporary fix until we fix our custom modal for Angular 1.5.

File: eucaconsole/static/js/pages/bucket_details.js (hunk 2 above)
Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: I'm considering converting the CORS config modal to a Foundation modal... thoughts?

File: eucaconsole/static/js/pages/bucket_details.js [hunk 4]
```diff
@@ -190,6 +188,29 @@ angular.module('BucketDetailsPage',
...
+                            styleActiveLine: true,
+                            autoCloseTags: true,
+                            lineNumbers: true
+                        });
+                    }
+                    // Reset to sample CORS config when re-adding post-deletion
+                    if (!$scope.hasCorsConfig && !!$rootScope.codeEditor) {
+                         $rootScope.codeEditor.setValue($scope.sampleCorsConfig);
+                    }
+                }, 300);
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: Did you try with no timeout? What that does is ensure this code runs after the current $apply(). Having a timeout makes me think "race condition" and that can be unreliable. An order change by simply making this run after the current $apply() is more sound a solution.

Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_details.js]: Looks like it works, I tried it. I did see that the dialog doesn't go away after saving new cors config. new ticket?

Comment (kamalgill) [inline eucaconsole/static/js/pages/bucket_details.js]: We need the $timeout (with a timeout value) for CodeMirror to work.  Removing that timeout breaks CodeMirror and causes the dialog to not go away after saving the CORS config.  The dialog works fine with the code as-is AFAICT.

**Pull request merged**